doc/en/developer/source/eclipse-guide/index.rst

```diff
@@ -18,6 +18,16 @@ Running and debugging
 Run or debug the class ``org.geoserver.web.Start`` in the ``web-app`` 
 module. The steps to do so are detailed in the :ref:`quickstart`.
 
+Running GeoServer with Extensions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, GeoServer will run without any extensions enabled. In order to run GeoServer with extensions, the ``web-app`` module declares a number of profiles used to enable specific extensions when running ``Start``. To enable an extension, re-generate the ``web-app`` eclipse profile with the appropriate maven profile(s) enabled::
+
+  % cd web/app
+  % mvn eclipse:eclipse -P wps
```

Comment:
Christian, I did that in the past but it occasionally caused issues because the project imported depended on the jars of other core projects that are also linked as project dependencies in web-app. Depending on the final classpath order you might end up wondering why a change made on the source (without a maven rebuild) does not get reflected in your debug sessions.